nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/main/java/org/nuxeo/ecm/platform/rendition/lazy/AbstractLazyCachableRenditionProvider.java

```diff
@@ -162,6 +158,143 @@ protected String toHexString(byte[] data) {
         return buf.toString();
     }
 
+    protected TransientStore getTransientStore() {
+        TransientStoreService tss = Framework.getService(TransientStoreService.class);
+        TransientStore ts = tss.getStore(CACHE_NAME);
+        if (ts == null) {
+            throw new NuxeoException("Unable to find Transient Store  " + CACHE_NAME);
+        }
+        return ts;
+    }
+
+    protected List<Blob> handleNewRendition(String key, DocumentModel doc, RenditionDefinition definition,
+            String sourceDocumentModificationDate) {
+        Work work = getRenditionWork(key, doc, definition);
+        if (log.isDebugEnabled()) {
+            log.debug(String.format(
+                    "No entry found for key %s in the %s transient store, scheduling rendition work with id %s and storing an empty blob for now.",
+                    key, CACHE_NAME, work.getId()));
+        }
+        if (sourceDocumentModificationDate != null) {
+            getTransientStore().putParameter(key, SOURCE_DOCUMENT_MODIFICATION_DATE_KEY,
+                    sourceDocumentModificationDate);
+        }
+        StringBlob emptyBlob = new StringBlob("");
+        emptyBlob.setFilename(LazyRendition.IN_PROGRESS_MARKER);
+        emptyBlob.setMimeType("text/plain;" + LazyRendition.EMPTY_MARKER);
+        getTransientStore().putBlobs(key, Collections.singletonList(emptyBlob));
+        Framework.getService(WorkManager.class).schedule(work, Scheduling.IF_NOT_SCHEDULED);
+        return getTransientStore().getBlobs(key);
```

Comment:
Why not return `Collections.singletonList(emptyBlob)` here? If going through the transient store is important then please add a comment explaining why.

Reply:
Because I kept the previous behavior added in https://github.com/nuxeo/nuxeo/commit/77a835fef01374b2d873a29f9b9a4612eba77acd#diff-46701308c6beebe73cc529828c24390eR87 :smile: ([NXP-20716](https://jira.nuxeo.com/browse/NXP-20716): Prevent long-running lazy rendition w/fast-changing source doc from starving others).

I assume this was to handle the case where the work is immediately executed and completed after the call to `WorkManager#schedule`, thus allowing to re-fetch an updated blob from the transient store, is that right? Though it seems surprising...

Reply:
I would mean the work is synchronous but it's never the case... So the original code was already doing unnecessary work going through the transient store.

Reply:
:+1: